CPU Spy/glue/ViewController.swift

```diff
@@ -9,25 +9,58 @@
 import Cocoa
 
 class ViewController: NSViewController {
+    let userDefaults = NSUserDefaults.standardUserDefaults()
+    let noteCenter = NSNotificationCenter.defaultCenter()
 
-    @IBOutlet var sampleIntervalForeground: NSTextField?
+    var runMode: RunMode = .Foreground
 
-    @IBAction func sampleIntervalForegroundChanged(sender: NSTextField) {
-        if NSUserDefaults.standardUserDefaults()
-            .doubleForKey(settingSampleIntervalForeground) == sender.doubleValue {
-            return
-        }
-        NSNotificationCenter.defaultCenter()
-            .postNotificationName(msgNewSampleIntervalForeground, object: sender.doubleValue)
-    }
+    @IBOutlet var sampleIntervalForeground: NSTextField?
+    @IBOutlet var sampleIntervalBackground: NSTextField?
+    @IBOutlet var sampleIntervalHidden: NSTextField?
 
     var processTableViewController: ProcessTableViewController! = ProcessTableViewController()
+
     @IBOutlet var processTableView: NSTableView? {
         didSet {
             processTableViewController.processTable = processTableView
         }
     }
 
+
+    // MARK: Appearance control
+    override func viewWillAppear() {
+        super.viewWillAppear()
+
+        // read settings from UserDefaults
+        let ud = userDefaults
+        sampleIntervalForeground?.doubleValue = ud.doubleForKey(settingSampleIntervalForeground)
+        sampleIntervalBackground?.doubleValue = ud.doubleForKey(settingSampleIntervalBackground)
+        sampleIntervalHidden?.doubleValue     = ud.doubleForKey(settingSampleIntervalHidden)
+
```

Comment:
What's the runMode at this point? Figure out way, to get an appropriate answer to this.